forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +272,133 @@ impl WalletCommands {
         }
     }
 }
+
+fn format_balance_string(num: Decimal, mode: FormattingMode) -> String {
+    let units = ["atto", "femto", "pico", "nano", "micro", "milli", ""];
+    let orig = num;
+
+    let mut num = num;
+    let mut unit_index = 0;
+    while num >= dec!(1000.0) && unit_index < units.len() - 1 {
+        num /= dec!(1000.0);
+        unit_index += 1;
+    }
+
+    match mode {
+        FormattingMode::ExactFixed => {
+            let fil = orig / dec!(1e18);
+            format!("{fil} FIL")
+        }
+        FormattingMode::NotExactFixed => {
+            let fil_orig = orig / dec!(1e18);
+            let fil = fil_orig.round_dp_with_strategy(3, RoundingStrategy::MidpointAwayFromZero);
+            let mut res = format!("{} FIL", fil.trunc());
+            if fil != fil_orig {
+                res.insert(0, '~');
+            }
+            res
+        }
+        FormattingMode::ExactNotFixed => format!("{num:0} {} FIL", units[unit_index]),
+        FormattingMode::NotExactNotFixed => {
+            let mut fil = num.round_dp_with_strategy(3, RoundingStrategy::MidpointAwayFromZero);
+            if fil == fil.trunc() {
+                fil = fil.trunc();
+            }
+            let mut res = format!("{} {} FIL", fil, units[unit_index]);
+
+            if fil != num {
+                res.insert(0, '~');
+            }
+
+            res
+        }
+    }
+}
+
+fn bool_pair_to_mode(exact: bool, fixed: bool) -> FormattingMode {
+    if exact && fixed {
+        FormattingMode::ExactFixed
+    } else if !exact && fixed {
+        FormattingMode::NotExactFixed
+    } else if exact && !fixed {
+        FormattingMode::ExactNotFixed
+    } else {
+        FormattingMode::NotExactNotFixed
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use fvm_shared::econ::TokenAmount;
+
+    use super::*;
+
+    #[test]
+    fn exact_balance_fixed_unit() {
+        let cases_vec = vec![
+            (100, "0.0000000000000001 FIL"),
+            (12465, "0.000000000000012465 FIL"),
+        ];
+
+        for (atto, result) in cases_vec {
+            test_call(atto, result, true, true);
+        }
+    }
+
+    #[test]
+    fn not_exact_balance_fixed_unit() {
+        let cases_vec = vec![
+            (100i64, "~0 FIL"),
```

Comment:
All those `i64` suffixes are not needed anymore.

Reply:
Fixed